cli/src/commit_templater.rs

```diff
@@ -2197,3 +2220,41 @@ pub fn builtin_cryptographic_signature_methods<'repo>(
     );
     map
 }
+
+#[derive(Debug, Clone)]
+pub struct AnnotateData {
+    pub commit: Commit,
+    pub line_no: usize,
+    pub is_first_line: bool,
+}
+
+pub fn builtin_annotate_data_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, AnnotateData>
+{
+    type L<'repo> = CommitTemplateLanguage<'repo>;
+    let mut map = CommitTemplateBuildMethodFnMap::<AnnotateData>::new();
+    map.insert(
+        "commit",
+        |_language, _diagnostics, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let out_property = self_property.and_then(|data| Ok(data.commit.clone()));
+            Ok(L::wrap_commit(out_property))
+        },
+    );
+    map.insert(
+        "line_no",
+        |_language, _diagnostics, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let out_property = self_property.and_then(|data| Ok(data.line_no.try_into()?));
+            Ok(L::wrap_integer(out_property))
+        },
+    );
+    map.insert(
+        "is_first_line",
```

Comment:
"first line" is confusing, and template methods don't have `is_` prefix. Maybe it can be "first_line_in_hunk" or "first_hunk_line"? It's not technically a "hunk", but I don't have a better suggestion.